python/cudf/cudf/core/dataframe.py

```diff
@@ -4026,16 +4017,16 @@ def stack(self, level=-1, dropna=True):
         dtype: int64
         """
         assert level in (None, -1)
-        index_as_cols = self.index.to_frame(index=False)._columns
-        new_index_cols = libcudf.filling.repeat(index_as_cols, self.shape[1])
-        [last_index] = libcudf.filling.tile(
-            [column.as_column(self.columns)], self.shape[0]
-        )
-        new_index_cols.append(last_index)
-        index_df = DataFrame(
-            dict(zip(range(0, len(new_index_cols)), new_index_cols))
+        repeated_index = self.index.repeat(self.shape[1])
+        repeated_index = [c for c in repeated_index._columns]
```

Comment:
Could we just update this to:
```suggestion
        repeated_index = [c for c in self.index._columns]
```
Since `.repeat` repeats vertically and not horizontally it shouldn't affect the columns that are output, no?

Reply:
I'm not sure what you're saying. `repeat` is not an inplace operation, so it creates entirely new columns / frame.

Reply:
I had a total brainfart and read this is `._column_names`. I'm losing it 😅.

Reply:
Oh, I see what you're saying. Writing response...